CHANGELOG.md

```diff
@@ -4,6 +4,9 @@ CHANGELOG
 ## HEAD (unreleased)
 - Fix handling of `nil` values in Outputs in Go.
   [#4268](https://github.com/pulumi/pulumi/pull/4268)
+  
+- Fix handling of secret values in mock-based tests.
+  [#4272](https://github.com/pulumi/pulumi/pull/4272)
```

Comment:
Do we have any tests that validate mocks working with secrets?

Reply:
I had the same question but it looks like we don't have any tests for mocks. Probably worth adding them for all languages / methods / parameter permutations, but sounds a bit out of the scope for this issue?

Reply:
We did add tests for Python mocks recently - https://github.com/pulumi/pulumi/pull/4074. 

But happy to split out a new issue on getting test coverage broadly on mocking.

Reply:
Awesome! I missed that one. Now I augmented the Python test to use a secret. Also created #4276 to add tests for other runtimes.